google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -1327,4 +1329,54 @@ public void floatIncrement() throws ExecutionException, InterruptedException {
       this.error = error;
     }
   }
+
+  @Test
+  public void getAllWithObserver() throws Exception {
+    DocumentReference ref = randomColl.document("doc1");
+    ref.set(ALL_SUPPORTED_TYPES_MAP).get();
+
+    DocumentReference ref1 = randomColl.document("doc2");
+    ref1.set(ALL_SUPPORTED_TYPES_MAP).get();
+
+    DocumentReference ref2 = randomColl.document("doc3");
+    ref2.set(ALL_SUPPORTED_TYPES_MAP).get();
+    ref2.delete();
+
+    final List<DocumentSnapshot> documentSnapshots = new ArrayList<>();
+    final DocumentReference[] documentReferences = {ref, ref1, ref2};
+    final SettableApiFuture<Void> future = SettableApiFuture.create();
+    firestore.getAll(
+        documentReferences,
+        FieldMask.of("foo"),
+        new ApiStreamObserver<DocumentSnapshot>() {
+
+          @Override
+          public void onNext(DocumentSnapshot documentSnapshot) {
+            documentSnapshots.add(documentSnapshot);
+          }
+
+          @Override
+          public void onError(Throwable throwable) {
+            future.setException(throwable);
+          }
+
+          @Override
+          public void onCompleted() {
+            future.set(null);
+          }
+        });
+
+    future.get();
+
+    if (null != documentSnapshots && documentSnapshots.size() > 0) {
```

Comment:
This if statement doesn't actually do anything. `documentSnapshots` is never null due to the final initializer above, and the size check before iterating is not needed due to the for-each not executing if the list is empty.

Reply:
done